extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -873,12 +875,36 @@ export class AmpA4A extends AMP.BaseElement {
     });
   }
 
+  /** @override **/
+  attemptChangeSize(newHeight, newWidth) {
+    // Store original size of slot in order to allow re-expansion on
+    // unlayoutCallback so that it is reverted to original size in case
+    // of resumeCallback.
+    this.originalSlotSize_ = this.getIntersectionElementLayoutBox();
+    super.attemptChangeSize(newHeight, newWidth).catch(() => {});
+  }
+
   /** @override  */
   unlayoutCallback() {
     // Increment promiseId to cause any pending promise to cancel.
     this.promiseId_++;
     this.protectedEmitLifecycleEvent_('adSlotCleared');
     this.uiHandler.setDisplayState(AdDisplayState.NOT_LAID_OUT);
+    if (this.originalSlotSize_) {
+      // Attempt to revert any size change, this could fail but is unlikely as
+      // we can assume the document is no longer visible.
+      protectFunctionWrapper(
```

Comment:
This crazy thing is useless. `#attemptChangeSize` will never throw an error, but it might return a rejected promise which you're not handling.